integrations/python/pymongo/workload-executor.py

```
@@ -35,92 +22,66 @@ def interrupt_handler(signum, frame):
-    # Do not modify connection string and do not add any extra options.
-    client = connect(mongodb_uri)
-
-    # Create test entities.
-    database = client.get_database(test_workload["database"])
-    collection = database.get_collection(test_workload["collection"])
-    objects = {"database": database, "collection": collection}
-
-    # Run operations
-    num_failures = 0
-    num_errors = 0
-    num_operations = 0
-    global IS_INTERRUPTED
-
-    operations = test_workload["operations"]
-    ops = [prepare_operation(op) for op in operations]
-
-    while True:
-        if IS_INTERRUPTED:
-            break
-        for op in ops:
-            try:
-                run_operation(objects, op)
-            except AssertionError:
-                traceback.print_exc()
-                num_failures += 1
-            except Exception:
-                traceback.print_exc()
-                num_errors += 1
-            else:
-                num_operations += 1
-
-    # We reach here once IS_INTERRUPTED has been set to True.
-    stats = {"numErrors": num_errors, "numFailures": num_failures,
-             "numSuccesses": num_operations}
-    print("Workload statistics: {!r}".format(stats))
-
-    sentinel = os.path.join(os.path.abspath(os.curdir), 'results.json')
-    print("Writing statistics to sentinel file {!r}".format(sentinel))
-    with open('results.json', 'w') as fr:
-        json.dump(stats, fr)
-    exit(0 or num_errors or num_failures)
+    runner = UnifiedSpecTestMixinV1()
+    runner.TEST_SPEC = test_workload
+    UnifiedSpecTestMixinV1.TEST_SPEC = test_workload
+    runner.setUpClass()
+    runner.setUp()
+    # should be removed in final version, but useful for testing right now
+    for op in test_workload["tests"][0]["operations"]:
+        if op.get("name") == "loop":
+            op["arguments"]["numIterations"] = 10
+    try:
+        runner.run_scenario(test_workload["tests"][0], uri=mongodb_uri)
```

We should assert that test_workload["tests"] only has one test in it to make sure we're not silently ignoring extra tests.